components/ai_chat/core/browser/engine/oai_api_client_unittest.cc

```diff
@@ -195,4 +194,65 @@ TEST_F(OAIAPIUnitTest, PerformRequest) {
   testing::Mock::VerifyAndClearExpectations(mock_request_helper);
 }
 
+TEST_F(OAIAPIUnitTest, HandleInavlidResponseFormat) {
+  mojom::CustomModelOptionsPtr model_options = mojom::CustomModelOptions::New(
+      "test_api_key", 0, 0, 0, "test_system_prompt", GURL("https://test.com"),
+      "test_model");
+
+  // Simulate an invalid response format from the server
+  std::string invalid_server_response =
+      R"({"choices":[{"message":{"not_content":"value"}}]})";
```

Comment:
can we add a couple more invalid test cases like

`aaaaaaaaaaaaaaaaa`
`{choices: []}`
`{"choices": [{"message": {"content": []}]}`

Reply:
and also test empty server response body

Reply:
In general, every check in the C++ code should have test case covered.